plugins/osdn/factory/factory.go

```diff
@@ -3,31 +3,31 @@ package factory
 import (
 	"strings"
 
-	"github.com/openshift/openshift-sdn/plugins/osdn"
-	"github.com/openshift/openshift-sdn/plugins/osdn/api"
-
 	osclient "github.com/openshift/origin/pkg/client"
 	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/storage"
 
+	"github.com/openshift/openshift-sdn/plugins/osdn"
+	"github.com/openshift/openshift-sdn/plugins/osdn/api"
 	"github.com/openshift/openshift-sdn/plugins/osdn/ovs"
 )
 
 // Call by higher layers to create the plugin SDN master instance
-func NewMasterPlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client) (api.OsdnPlugin, error) {
-	return newPlugin(pluginType, osClient, kClient, "", "")
+func NewMasterPlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client, etcdHelper storage.Interface) (api.OsdnPlugin, error) {
```

Comment:
We can't directly use the API in this case. As the existing NetNamespace resource is converted to virtual resource backed by namespace to support backward compatibility, we need the etcd access to directly access /registry/sdnnetnamespaces for migration.